ops/testing.py

```diff
@@ -284,6 +284,9 @@ def charm(self) -> CharmType:
 
         Note that the Charm is not instantiated until :meth:`.begin()` is called.
         Until then, attempting to access this property will raise an exception.
+
+        Raises:
+            RuntimeError: if :meth:`begin` has not been called.
```

Comment:
Hmm, I'm not sure about specifying all these `RuntimeError`s. I think we don't necessarily want to make the exception type part of the contract. They're almost all a programmer error so should fail hard, rather than the caller catching them. Agreed?